### ISSUE TITLE
Allow more controll over generated values

### ISSUE THREAD

**Issue by PascalSchumacher** [benas/random-beans#118]
Related to https://github.com/benas/random-beans/issues/113 where stringswithout chinese characters were requested I think it could be a good idea to allow more control over generated values.

Maybe we can take some inspiration from https://github.com/DiUS/fatboy, which uses:

```java
Configurer.configure()
    .collections(FieldLength.random(2, 5))
    .strings(FieldLength.random(10, 25))
    .integers(FieldLength.random(0, 100000))
```

and extend it like

```java
.dates(min, max)
.characters(charset)
```
?

**Comment by benas** [comment]
Hi, cool project, similar to random beans.

> I think it could be a good idea to allow more control over generated values

This is currently possible and can be done through the registration of custom randomizers (DateRangeRandomizer, IntegerRangeRandomizer, etc). What is missing is probably adding more options to some existing randomizers (like charset to CharacterRandomizer and StringRandomizer).

Do you think we really need to add it to the builder API?

**Comment by PascalSchumacher** [comment]
Hi Mahmoud,

I completely missed this option. As you said it's already possible with 3.0, e.g.:

```java
EnhancedRandomBuilder.randomize(Date.class, new DateRangeRandomizer(min, max)).randomize(Integer.class, new IntegerRangeRandomizer(10, 1000))
```

I think this option is fine, no need to add it to the builder API.

As you said there could be further options for some randomizers and some additional RangeRandomizers e.g. for LocalDateTime.

**Comment by benas** [comment]
I do agree, I think the current option is enough for 3.0.

I'll make `StringRandomizer` and `LocalDateTime` randomizer more configurable (and may be other randomizers). The `CharacterRandomizer` can be changed in your PR #119 . Is this ok for you?

**Comment by benas** [comment]
The `CharacterRandomizer` delegates to the `StringRandomizer`, we can not work in parallel arrgh :rage: I like your `collectPrintableCharactersOf` and `isPrintable` methods :+1: I want to use them in the ``StringRandomizer`. Could you move them to an utility `CharacterUtils` class?

**Comment by PascalSchumacher** [comment]
>if only we could cherry pick from a PR 

It's possible to (kind of) cherry-pick from a PR, but it's a bit difficult.

Cherry-pick the changes locally. Refactor. Amend the last commit and but keep the original author.

**Comment by benas** [comment]
fixed in caa5e714603f543ffc25c7dd8d8b9abf8c01be5c and 32d24678484f2126369a34ab9c79847a486fc985
